src/Confluent.Kafka/Producer.cs

```diff
@@ -548,12 +613,35 @@ public void HandleDeliveryReport(Message message)
             }
         }
 
-        public void ProduceAsync(string topic, TKey key, TValue val, IDeliveryHandler<TKey, TValue> deliveryHandler, DateTime? timestamp = null, int? partition = null, bool blockIfQueueFull = true)
+        private void Produce(string topic, TKey key, TValue val, IDeliveryHandler<TKey, TValue> deliveryHandler, DateTime? timestamp, int partition, bool blockIfQueueFull)
         {
             var handler = new TypedDeliveryHandlerShim(key, val, deliveryHandler);
-            producer.ProduceAsync(topic, KeySerializer?.Serialize(key), ValueSerializer?.Serialize(val), handler, timestamp, partition, blockIfQueueFull);
+            var keyBytes = KeySerializer?.Serialize(key);
```

Comment:
hi @tomasdeml, thanks for pointing that out.. 

the last few PRs were pushed without sufficient testing, sorry about this. FWIW this is fixed in the prodtest branch.

note that 0.9.2.x (currently == master) is what we intend to release (+ bug fixes). we are now completely dedicated to testing.